### PR TITLE
Remove using empty base AO ctor

### PR DESCRIPTION
**Pull request by DmitryGorokhov**
Resolve comment from pr #6432 

## Proposed changes

- Remove call empty ctor of Accessible Object class in inherited classes.

## Customer Impact

- No

## Regression? 

- No

## Risk

- No

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6571)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.TopRowAccessibleObject.cs [hunk 1]
```diff
@@ -16,11 +16,11 @@ protected class DataGridViewTopRowAccessibleObject : AccessibleObject
             private int[]? runtimeId;
             private DataGridView? _ownerDataGridView;
 
-            public DataGridViewTopRowAccessibleObject() : base()
+            public DataGridViewTopRowAccessibleObject()
             {
             }
```
Comment (WPMGPRoSToTeMa) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.TopRowAccessibleObject.cs:21]: Can we also remove the parameterless constructor?

**Review by WPMGPRoSToTeMa — APPROVED** [review]

**Comment by WPMGPRoSToTeMa** [comment]
It's safe to remove `: base()` even when it calls non-empty parameterless constructor because it's called implicitly:
> In a derived class, if a base-class constructor is not called explicitly by using the base keyword, the parameterless constructor, if there is one, is called implicitly.
https://docs.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/using-constructors

The only question whether we want be explicit about calling base parameterless constructors or not.

cc: @RussKie

**Review by vladimir-krestov — DISMISSED** [review]
Approved ✔️

**Comment by RussKie** [comment]
The default .ctors are called implicitly if those are present and no other .ctor is explicitly called.

**Review by RussKie — COMMENTED** [review]
:shipit: once the comment is addressed.

**Comment by WPMGPRoSToTeMa** [comment]
> The default .ctors are called implicitly if those are present and no other .ctor is explicitly called.

I mean is there a code style rule that will warn when the base parameterless .ctor is called explicitly? (to avoid this in future)

**Review by vladimir-krestov — DISMISSED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]

**Comment by RussKie** [comment]
> > The default .ctors are called implicitly if those are present and no other .ctor is explicitly called.
> 
> I mean is there a code style rule that will warn when the base parameterless .ctor is called explicitly? (to avoid this in future)

The compiler will tell you :)

If there is no .ctor defined - one is implicitly added and called. If there is a non-default .ctor (i.e. with parameters) - it will always be called, no default .ctor will be generated.

File: src/System.Windows.Forms/src/PublicAPI.Shipped.txt [hunk 2]
```diff
@@ -6816,7 +6816,6 @@ System.Windows.Forms.DataGridView.DataGridViewAccessibleObject
 System.Windows.Forms.DataGridView.DataGridViewAccessibleObject.DataGridViewAccessibleObject(System.Windows.Forms.DataGridView! owner) -> void
 System.Windows.Forms.DataGridView.DataGridViewControlCollection
 System.Windows.Forms.DataGridView.DataGridViewTopRowAccessibleObject
-System.Windows.Forms.DataGridView.DataGridViewTopRowAccessibleObject.DataGridViewTopRowAccessibleObject() -> void
```
Comment (RussKie) [inline src/System.Windows.Forms/src/PublicAPI.Shipped.txt:6819]: We must not alter the public API surface. See https://docs.microsoft.com/dotnet/api/system.windows.forms.datagridview.datagridviewtoprowaccessibleobject.-ctor

**Review by RussKie — CHANGES_REQUESTED** [review]

**Comment by WPMGPRoSToTeMa** [comment]
> 
> The compiler will tell you :)
> 
> If there is no .ctor defined - one is implicitly added and called. If there is a non-default .ctor (i.e. with parameters) - it will always be called, no default .ctor will be generated.

This is what I actually meant:
```
public DataGridViewTopRowAccessibleObject() : base()
                                              ^~~~~~
                                              |
                                              Warning: base parameterless constructor is called explicitly, please don't do that
```
And as I can see there is no rule for that :(

File: src/System.Windows.Forms/src/PublicAPI.Shipped.txt (hunk 2 above)
Comment (DmitryGorokhov) [reply src/System.Windows.Forms/src/PublicAPI.Shipped.txt]: Ok, restored paramless ctor and tis file. Thanks

**Review by RussKie — APPROVED** [review]

**Pull request merged**